src/lexer/diagnostics.rs

```diff
@@ -2,6 +2,6 @@ use miette::{self, Diagnostic};
 use thiserror::{self, Error};
 
 #[derive(Debug, Error, Diagnostic)]
-#[error("Syntax error")]
+#[error("Syntax error {0}")]
```

Comment:
What does this change do?

Reply:
Prints out the given string to this error. Makes it possible to write

Syntax error err_str